### ISSUE TITLE
Refactor diffusion testnet simulation

### ISSUE THREAD

**Issue by coot** [input-output-hk/ouroboros-network#4085]
The current module structure and some abstractions makes things unnecessarily
complicated.  I think we should:

* merge parts of `Test.Ouroboros.Network.Diffusion.Node` and `Test.Ouroboros.Network.Testnet.Simulation.Node` which are used to run a single diffusion, exposing only the simpler interface (from the latter module);
* split Test.Ouroboros.Network.Testnet.Simulation.Node into two parts:

    - `Test.Ouroboros.Network.Diffusion.Testnet` which runs a network
    - `Test.Ouroboros.Network.Diffusion.Types` which includes `NodeArgs`, `SimArgs` and their arbitrary generators;

* use `Test.Ouroboros.Network.Diffusion` as the module which exposes all the
  tests.

**Comment by coot** [comment]
Extra points for more relaxed imports, e.g.
```
import           Test.Ouroboros.Network.Testnet.Simulation.Node
```
instead
```
import           Test.Ouroboros.Network.Testnet.Simulation.Node (Command (..),
                     DNSLookupDelay (..), DNSTimeout (..), DiffusionScript (..),
                     DiffusionSimulationTrace (..),
                     HotDiffusionScript (HotDiffusionScript), NodeArgs (..),
                     SimArgs (..), TestAddress (..), diffusionSimulation,
                     prop_diffusionScript_commandScript_valid,
                     prop_diffusionScript_fixupCommands)
```
which makes it much easier to paste and try quickcheck coutnerexamples.

**Comment by coot** [comment]
Is it possible to remove `WithTime`? `IOSim` already includes time in its trace.

**Comment by bolt12** [comment]
> Is it possible to remove `WithTime`? `IOSim` already includes time in its trace.

Yes it is. IIRC `WithTime` was being used to take 10h of execution but that got changed to just inspect 125k events

**Comment by coot** [comment]
> Extra points for more relaxed imports

That's done in #4086 :).
